### ISSUE TITLE
Energy sensors dont work properly in Bosch Thermostat v0.22.2, device CT200

### ISSUE THREAD

**Issue by mcervar** [bosch-thermostat/home-assistant-bosch-custom-component#291]
Hi

After upgrading to version v0.22.2 energy sensor stopped working properly.
Either per day data is missing, or values are in negative on energy dashboard.

HA version is 2023.4.3
Device is CT200

![image](https://user-images.githubusercontent.com/47822784/231779324-dd7a9afd-f290-4f00-bc8c-961641d7de14.png)

![image](https://user-images.githubusercontent.com/47822784/231779502-b750fb9b-71a6-4361-a8e5-76698ddb0277.png)

Not sure what to do or how to properly debug, any help is welcome :D

**Comment by pszafer** [comment]
Do you have New API enabled?
If not I haven't changed anything in "regular" algorithm, but it shouldn't be used for energy stats. CT200 provide only total last day of energy, so it can't be used that way.

**Comment by TaxmanNo1** [comment]
Hello,
i´ve got the same issue as @mcervar. 

I´ve deleted the integration and reinstalled it. 
The new API is enabled but
The recording entities are all showing the value "unknown"

My device is RC300 / RC310 
All updates are installed

**Comment by mcervar** [comment]
Yes, with this new version first I deleted the integration and entities if lift, installed new one and enabled new api :/

I can do it again but I doubt it will help

**Comment by wernerkarlheisenberg** [comment]
I've got the same issue as @mcervar, I'm also using the CT200.

I also deleted the integration and reinstalled it.
The new API is enabled.

In the past it worked fine. It is true that there is only an aggregated value per day, but in the past this value was divided by 24 and spread throughout the whole day.

![image](https://user-images.githubusercontent.com/20357407/233605457-56b96fae-9c77-4754-8864-45aeb3dc8843.png)

**Comment by mcervar** [comment]
Just update how my energy gas consumption chart looks like

![image](https://user-images.githubusercontent.com/47822784/233920279-3edaff90-bb22-4699-b49e-806aca987947.png)

**Comment by TaxmanNo1** [comment]
So... A little Workaround:
Just deleted the bosch thermostat integration and the hacs repository and reinstalled everything with version 20.2. Everything working fine again.

**Comment by clemenslermen** [comment]
Same issue here

**Comment by mcervar** [comment]
Tried deleting bosch thermostat integration and the hacs repository and reinstalled everything with version 20.2, didnt help
What helped a bit is going to Developers tools --> Statistics --> finding energy sensors and setting values manually
But again once day passes and new data should be shown, its not shown correctly or it is the same one as from day before, which is not possible

**Comment by wernerkarlheisenberg** [comment]
I figured out I had multiple sensors that I could select for the energy dashboard. While sensors.energy_hot_water provided wrong (even negative) values as mentioned above, sensors.ehwenergy_hot_waterexternal did provide correct values. At a certain point in time data was no longer updated, but when I enabled debug mode it started to update again every night (and filled a value for the missed days). Could be that enabling debug mode is not necessary and a simple restart of HA works as well ...
